contracts/allocators/OnsenAllocatorV2.sol

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.10;
+import "../libraries/Address.sol";
+
+// types
+import "../types/BaseAllocator.sol";
+
+struct UserInfo {
+    uint256 amount; // How many LP tokens the user has provided.
+    uint256 rewardDebt;
+}
+
+interface IMasterChef {
+    function pendingSushi(uint256 _pid, address _user) external view returns (uint256);
+
+    function deposit(uint256 _pid, uint256 _amount) external;
+
+    function withdraw(uint256 _pid, uint256 _amount) external;
+
```

Comment:
Theres a subfolder for interfaces in the allocators folder. Think this belongs there.

Reply:
Interfaces moved to the interface subfolder and the contract was updated to reference the interfaces.